tests/backup/backup_test.go

```diff
@@ -7407,3 +7408,147 @@ var _ = Describe("{ScheduleBackupCreationAllNS}", func() {
+		backupLocationName string
+		deploymentName     string
+		restoreName        string
+	)
+	labelSelectors := make(map[string]string)
+	namespaceMapping := make(map[string]string)
+	newBackupLocationMap := make(map[string]string)
+	bkpNamespaces = make([]string, 0)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("CustomResourceRestore", "Create custom resource restore", nil, 58041)
+		log.InfoD("Deploy applications")
+
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Create custom resource restore", func() {
+		providers := getProviders()
+		Step("Validate applications", func() {
+			log.Infof("Validate applications")
+			ValidateApplications(contexts)
+		})
+		Step("Create User", func() {
+			log.InfoD("Creating user")
+			userName = createUsers(1)
+			log.Infof("Created user is %v", userName)
+
+		})
+		Step("Adding Credentials and Registering Backup Location", func() {
+			log.Infof("Creating cloud credentials and backup location")
+			for _, provider := range providers {
+				cloudCredUID = uuid.New()
+				CloudCredUidList = append(CloudCredUidList, cloudCredUID)
+				backupLocationUID = uuid.New()
+				credName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				CreateCloudCredential(provider, credName, cloudCredUID, orgID)
+				log.InfoD("Created Cloud Credentials with name - %s", credName)
+				time.Sleep(time.Minute * 1)
+				backupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				newBackupLocationMap[backupLocationUID] = backupLocationName
+				err := CreateBackupLocation(provider, backupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				log.FailOnError(err, "Creating Backup location %v", backupLocationName)
+				log.InfoD("Created Backup Location with name - %s", backupLocationName)
+			}
+		})
+		Step("Register source and destination cluster for backup", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			log.FailOnError(err, "Creating Source and destination cluster")
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying backup cluster %s", SourceClusterName))
+
+		})
+		Step("Taking backup of applications", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			namespace := bkpNamespaces[0]
```

Comment:
How are you making sure that your application namespace is selected in case we run this in the pipeline with multiple apps?

Reply:
I've added the for loop in case of multiple apps

Reply:
yes in the latest commit I've added the loop which will take care of multiple apps